### PR TITLE
Color cycle specviz

### PR DESCRIPTION
**Pull request by bmorris3**
### Description

This pull request addresses a specviz use case with various spectral orders, where all spectral orders may be plotted in the same specviz instance and viewed simultaneously, like in echelle spectroscopy. By default, `specviz.load_spectrum` will load and plot _every_ spectrum with a black line, which makes overlapping spectral orders quite difficult to visually disentangle, like this: 

bqplot (12)

In this PR, I revived a draft PR by @kecnry (#988) which implements a simple color cycler. The only change I've made to @kecnry's implementation is to use matplotlib's default color palette.

A similar plot after the changes in this PR looks like this:
 
bqplot (1)

The user can still manually adjust the colors via the plot options menu.

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set?
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Review by duytnguyendtn — COMMENTED** [review]
Much to the dismay of @pllim, we actually list `matplotlib` as a dependency for `jdaviz`

https://github.com/spacetelescope/jdaviz/blob/main/setup.cfg#L23

Would it be more robust to pull these values from matplotlib directly, rather than hardcoding them here?

In addition, should we be sensitive to color-blind individuals here? @Jenneh might want to weigh in with an opinion

**Comment by kecnry** [comment]
It does seem that the blocker to #988 (https://github.com/spacetelescope/jdaviz/pull/974#issuecomment-987187116) is no longer an issue now that we control plot options ourselves, so I'm really excited to see this revived!  But I also remember a lot of debate about the colors themselves (whether the first layer should still be a dark gray, and whether to use jdaviz-themed colors which were used in #988)

**Comment by bmorris3** [comment]
> But I also remember a lot of debate about the colors themselves (whether the first layer should still be a dark gray, and whether to use jdaviz-themed colors which were used in https://github.com/spacetelescope/jdaviz/pull/988)

Should we loop in Jennifer Kotler?

File: jdaviz/configs/specviz/plugins/viewers.py [hunk 1]
```diff
@@ -377,6 +377,27 @@ def add_data(self, data, color=None, alpha=None, **layer_state):
             if "Subset" in layer.layer.label and layer.layer.data.label == data.label:
                 layer.linewidth = 3
 
+        # This default color cycle was retrieved from matplotlib with the following lines:
+        #   import matplotlib.pyplot as plt
+        #   plt.rcParams['axes.prop_cycle'].by_key()['color']
+
+        color_cycle = [
```
Comment (pllim) [inline jdaviz/configs/specviz/plugins/viewers.py:384]: Glue or matplotlib must already have one that we can use?

Comment (bmorris3) [reply jdaviz/configs/specviz/plugins/viewers.py:384]: Similar thoughts here: https://github.com/spacetelescope/jdaviz/pull/1674#issuecomment-1261130604

**Comment by bmorris3** [comment]
In the latest few commits on this PR, I've implemented a of the requested features from discussions with @kecnry and @Jenneh. They are: 

* start the new color cycler with the usual (glue default) dark gray so that single datasets still look like they did before this PR
* after first data have been loaded in dark gray, the matplotlib default color cycle is adopted
* these selections aren't hard-coded anymore, instead they're extracted from glue and matplotlib
* [**this bit needs careful review**]: the changes have not been moved so they're implemented on `jdaviz.Application._update_selected_data_items` rather than within `Specviz`. That way the changes are applied to any viewer that supports multiple loads. The default behavior for single loads is unchanged (see first bullet point).

Screen Shot 2022-09-29 at 08 47 33

To emphasize: I'm still learning the many implications of implementing something on `jdaviz.Application` rather than within a specific viz/viewer, so the last bullet point may cause issues to spring up elsewhere, but I haven't seen it break anything yet. Advice appreciated!

**Review by kecnry — APPROVED** [review]
Doesn't seem to cause any issues in the other viztools (I was a bit concerned about mosviz getting very colorful on first load, but that doesn't seem to be the case).  I think this is worth getting in as-is and we can always iterate on the colors or consider a single cycler for both subsets and lines, if we find they clash too much with subset colors.

**Comment by kecnry** [comment]
Although since this now affects all viztools (except Imviz, I suppose), changing the PR title and tags might help future us

**Comment by pllim** [comment]
Ehem... did anyone try in dark mode? 😉

**Review by duytnguyendtn — APPROVED** [review]
Works well for me! Although I didn't check dark mode. I agree with Kyle this is a good increment and we can always go back and modify the actual colors later, if it calls on it.

**Comment by bmorris3** [comment]
@pllim here's a dark-mode preview:

Screen Shot 2022-09-29 at 11 58 10

**Pull request merged**